### PR TITLE
Modernise project & remove vulnerabilities

### PR DESCRIPTION
**Pull request by OllysCoding**
### What does this commit to do?

This commit modernises the project & brings it up-to-date with Nodejs & dependencies, removing all high vulnerabilities from the codebase

Refactors:
- Remove the import from google spreadsheet option. While useful, the mechanism for getting the authentication for this has changed, and its no longer useful to us.
- Adds an import from text file option to replace the functionality from importing from google spreadsheets
- Creates a local copy of 'merge-img', as the orginal project has not been updated, and was using vulnerable dependencies.
- Updates node.js to v16
- Updates entire project to use esmodules instead of common js
- Eliminates all current vulnerabilities from the project.

### Why do it this way?

There were a couple of key decisions made when doing this refactor:
1. **Removing google sheets integration**: After talking to @DavidLawes it was clear the original way authentication for this was intended to work was no longer available, given that the google sheets integration was an experiment, I replaced it with a simple text file parser instead.
2. **Bringing the merge-img library into the codebase**: The library itself hasn't been updated in sometime, and my look around the web I couldn't any 'drop in' solution that was up-to-date, faced with either writing the code myself, or copying the library across and updating the dependencies, I decided copying it should be fine. The code is small & only 3 files, so future maintenance shouldn't be much work (if this project is used at all)
3. **Updating from commonjs to esmodules**: I got myself a bit muddled when doing this so I'm not entirely sure it's needed, but overall I think the change is fine and helps modernise the codebase a bit.

### Testing

Testing this project was quite difficult, as we no longer have an active URLBox license, but I was able to test that
1. The text file parsing works
2. The URL box integration generates the correct requests (note: these requests never completed)
3. the `merge-img` integration correctly combines base64 encoded images from a buffer
4. The flow of all of these elements work together as expected

I **did not** test the ophan integration as the Ophan API seems to require authentication, so I've left that unchanged.

### CODE REVIEW

File: lib/csv.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import fs from 'fs'
+import { parse } from 'csv-parse/sync'
```
Comment (AshCorr) [inline lib/csv.js]: Considering the input file is just a URL on each line is it still worth including a CSV parser package to read it?

**Review by AshCorr — APPROVED** [review]
Fab!

File: lib/csv.js (hunk 1 above)
Comment (OllysCoding) [reply lib/csv.js]: This is a fair point, might do a quick refactor to just make it a read-from-txt file type deal

**Review by jamesgorrie — APPROVED** [review]
Great!

> no longer have a URLBox license

In the long term, if this tool is useful, we could swap to Percy perhaps? [There are ways to compare two URLs](https://docs.percy.io/docs/comparing-two-urls-to-each-other#:~:text=To%20compare%20two%20URLs%20to,test%20setup%20is%20very%20different) which might make this whole tool redundant. 

---

re: `merge-img` dependency - [there is a PR to tries to update the dependencies](https://github.com/preco21/merge-img/pull/16) with no luck, which refers to [another fork that does update them](https://www.npmjs.com/package/merge-img-vwv).

Given the size of this package, the route you've taken does feel more maintainable.

**Review by bryophyta — APPROVED** [review]
this looks good to me 👍 

though as per Ash's comment, it feels like dropping the csv dependency is a good call if it's not too time-consuming

**Comment by OllysCoding** [comment]
> In the long term, if this tool is useful, we could swap to Percy perhaps?

I think this is probably the way to go!

> it feels like dropping the csv dependency is a good call if it's not too time-consuming

Done :)

**Pull request merged**